src/librbd/internal.cc

```diff
@@ -2862,10 +2864,16 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
     }
 
     if (next_mirror_mode == cls::rbd::MIRROR_MODE_IMAGE) {
+      if (current_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
```

Comment:
Thanks, will prefix with note in next PR.

I can put this within CLI but its seems to be costly operation. because if i put this within execute_enable and execute_disable function than i need to initialize io_ctx for getting current_mirror_mode (cls_client::mirror_mode_get(&io_ctx, &current_mirror_mode)).  if i put this within execute_enable_disable than i need to compare it with two type of enum one is rbd_mirror_mode_t  and second one is cls::rbd::MirrorMode  which doesn't seems to be good because if in future if we change any one of the enum sequence than it might create problem. Any how it can be done at cli side

@dillaman  should i stick with this method or do within cli, thoughts ?